OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,76 +1,75 @@
+% (3) Rotating table data and writing to a new table
+% (4) Writing marker (.trc) and force (.mot) data to file
+
+% ----------------------------------------------------------------------- %
+% The OpenSim API is a toolkit for musculoskeletal modeling and           %
+% simulation. See http://opensim.stanford.edu and the NOTICE file         %
+% for more information. OpenSim is developed at Stanford University       %
+% and supported by the US National Institutes of Health (U54 GM072970,    %
+% R24 HD065690) and by DARPA through the Warrior Web program.             %
+%                                                                         %
+% Copyright (c) 2005-2016 Stanford University and the Authors             %
+% Author(s): James Dunne                                                  %
+%                                                                         %
+% Licensed under the Apache License, Version 2.0 (the "License");         %
+% you may not use this file except in compliance with the License.        %
+% You may obtain a copy of the License at                                 %
+% http://www.apache.org/licenses/LICENSE-2.0.                             %
+%                                                                         %
+% Unless required by applicable law or agreed to in writing, software     %
+% distributed under the License is distributed on an "AS IS" BASIS,       %
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         %
+% implied. See the License for the specific language governing            %
+% permissions and limitations under the License.                          %
+% ----------------------------------------------------------------------- %
+
+% Author: James Dunne, Shrinidhi K. Lakshmikanth, Chris Dembia, Tom Uchida,
+% Ajay Seth, Ayman Habib, Jen Hicks.
 
 import org.opensim.modeling.*
 
-
-
 %% Use a c3dAdapter to turn read a c3d file
-adapter = C3DFileAdapter()
+adapter = C3DFileAdapter();
 
-tables = adapter.read('test_walking.c3d')
+tables = adapter.read('test_walking.c3d');
 
-
-%% get the Markers
+%% get the marker data
 markers = tables.get('markers');
-
-markerdata = opensimTimeSeriesTableToMatlab(markers);
-markerlabels = fieldnames(markerdata);
-
-% Print the (unrotated) markers to trc file
-trcfileadapter = TRCFileAdapter();
-trcfileadapter.write(markers,'test_walking.trc');
-
-%% Get the force
-% forces = tables.get('forces');
-% forcedata = opensimVec3TableToStruct(forces);
-% markerlabels = fieldnames(forcedata);
-% 
-% % Print the (unrotated) markers to trc file
-% trcfileadapter = TRCFileAdapter();
-% trcfileadapter.write(markers,'test_walking.trc');
-
+% get the number of markers and frames
+nMarkers = markers.getNumColumns();
+nMarkerFrames = markers.getNumRows();
+% get the array of the metadata key names
+MarkerMetaDataKeys = markers.getTableMetaDataKeys();
+MarkerRate = str2num(markers.getTableMetaDataAsString(MarkerMetaDataKeys.get(0)));
```

Comment:
How would a user find out what index the marker rate is stored in? What's stored in the other keys? Could provide a list (or hyperlink) in the comment.